roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -147,17 +157,21 @@ def available_images(self, images, registries, task_vars):
         """Inspect existing images using Skopeo and return all images successfully inspected."""
         return [
             image for image in images
-            if any(self.is_available_skopeo_image(image, registry, task_vars) for registry in registries)
+            if self.is_available_skopeo_image(image, registries, task_vars)
         ]
 
-    def is_available_skopeo_image(self, image, registry, task_vars):
-        """Uses Skopeo to determine if required image exists in a given registry."""
+    def is_available_skopeo_image(self, image, registries, task_vars):
+        """Use Skopeo to determine if required image exists in known registry(s)."""
+
+        # if image does already includes a registry, just use that
+        if image.count("/") > 1:
+            registry, image = image.split("/", 1)
+            registries = [registry]
```

Comment:
The complication seem to arise from the fact that some images must come from a specific registry, while others come from a list of registries... hmm...

Would it make sense to (re)define the return value of `required_images` such that it can be used consistently without special-casing what has a fixed registry and what may come from different registries?

```python
{
    'image': {'registries': ['possible', 'registries']},
    'openshift/origin-pod:vtest': {'registries': ['docker.io', 'registry.access.redhat.com']}, # multiple registries possible
    'openshift3/registry-console': {'registries': ['registry.access.redhat.com']}, # must come from a specific registry
    # ...
}
```